app/lib/efile/md/md502_calculator.rb

```diff
@@ -465,6 +476,128 @@ def calculate_line_22b
         (@direct_file_data.fed_eic_qc_claimed && line_or_zero(:MD502_LINE_22).positive?) ? "X" : nil
       end
 
+      def calculate_line_27
+        # STUBBED PLEASE REPLACE
+        0
+      end
+
+      def calculate_line_28_local_tax_rate
+        case @intake.residence_county
+        when "Allegany", "Carroll", "Charles"
+          0.0303
+        when "Baltimore City", "Baltimore County", "Caroline", "Dorchester", "Howard", "Kent", "Montgomery", "Prince", "Queen Anne", "Somerset", "Wicomico"
+          0.0320
+        when "Calvert", "St. Mary"
+          0.0300
+        when "Cecil"
+          0.0275
+        when "Garrett"
+          0.0265
+        when "Harford"
+          0.0306
+        when "Talbot"
+          0.0240
+        when "Washington"
+          0.0295
+        when "Worcester"
+          0.0295
+        when "Anne Arundel"
+          0.027
+        when "Frederick"
+          taxable_net_income = line_or_zero(:MD502_LINE_20)
+          if filing_status_dependent? || filing_status_single? || filing_status_mfs?
+            if taxable_net_income <= 25_000
+              0.0225
+            elsif taxable_net_income <= 50_000
+              0.0275
+            elsif taxable_net_income <= 150_000
+              0.0296
+            else
+              0.032
+            end
+          elsif filing_status_mfj? || filing_status_hoh? || filing_status_qw?
+            if taxable_net_income <= 25_000
+              0.0225
+            elsif taxable_net_income <= 100_000
+              0.0275
+            elsif taxable_net_income <= 250_000
+              0.0296
+            else
+              0.032
+            end
+          end
+        end
+      end
+
+      def local_tax_rate
+        @lines[:MD502_LINE_28_LOCAL_TAX_RATE]&.value || 0
+      end
+
+      def calculate_line_28_local_tax_amount
+        taxable_net_income = line_or_zero(:MD502_LINE_20)
+
+        if @intake.residence_county == "Anne Arundel"
+          brackets = if filing_status_dependent? || filing_status_single? || filing_status_mfs?
+                       [
+                         { threshold: 50_000, rate: 0.0270 },
```

Comment:
How often are we using floats for tax math? I feel like maybe we should do `0.0270.to_d` instead when dealing with money to convert to a decimal.

Reply:
@powersurge360 thats a good question! My limited intuition is not that much since `line_or_zero` will turn everything to an integer; happy to convert these to decimals though